### PR TITLE
conditionally check for kde-release-key

### PR DESCRIPTION
**Pull request by carlosdem**
as it can't build itself the first time if it depends on itself already being installed

### CODE REVIEW

File: nci/lib/setup_repo.rb [hunk 1]
```diff
@@ -59,8 +59,11 @@ def setup_repo!(with_source: false, with_proxy: true, with_install: true)
     # Make sure we have the latest pkg-kde-tools, not whatever is in the image.
     return unless with_install
 
-    raise 'failed to install deps' unless Apt.install(%w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations kde-release-keyring])
-
+    @deps = %w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations])
```
Comment (hsitter) [inline nci/lib/setup_repo.rb:62]: there is an excess ) at the end

File: nci/lib/setup_repo.rb [hunk 2]
```diff
@@ -59,8 +59,11 @@ def setup_repo!(with_source: false, with_proxy: true, with_install: true)
...
+      @deps << 'kde-release-keyring' unless JOB_NAME.include?('kde-release-keyring')
```
Comment (hsitter) [inline nci/lib/setup_repo.rb:63]: is JOB_NAME actually defined here? I am thinking you may need to pull it out of `env`

File: nci/lib/setup_repo.rb (hunk 1 above)
Comment (carlosdem) [reply nci/lib/setup_repo.rb]: woops.

File: nci/lib/setup_repo.rb (hunk 2 above)
Comment (carlosdem) [reply nci/lib/setup_repo.rb:64]: done

File: nci/lib/setup_repo.rb [hunk 3]
```diff
@@ -59,8 +59,12 @@ def setup_repo!(with_source: false, with_proxy: true, with_install: true)
     # Make sure we have the latest pkg-kde-tools, not whatever is in the image.
     return unless with_install
 
-    raise 'failed to install deps' unless Apt.install(%w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations kde-release-keyring])
-
+    ENV.fetch('JOB_NAME')
```
[the remark this answers is not in the document]
Comment (carlosdem) [reply nci/lib/setup_repo.rb]: like so??

File: nci/lib/setup_repo.rb [hunk 4]
```diff
@@ -59,8 +59,11 @@ def setup_repo!(with_source: false, with_proxy: true, with_install: true)
     # Make sure we have the latest pkg-kde-tools, not whatever is in the image.
     return unless with_install
 
-    raise 'failed to install deps' unless Apt.install(%w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations kde-release-keyring])
-
+    @deps = %w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations]
+      @deps << 'kde-release-keyring' unless ENV.fetch('JOB_NAME').include?('kde-release-keyring')
+      raise 'failed to install deps' unless Apt.install(@deps)
+    end
```
Comment (hsitter) [inline nci/lib/setup_repo.rb:65]: There is a stray end here

File: nci/lib/setup_repo.rb [hunk 5]
```diff
@@ -59,8 +59,11 @@ def setup_repo!(with_source: false, with_proxy: true, with_install: true)
     # Make sure we have the latest pkg-kde-tools, not whatever is in the image.
     return unless with_install
 
-    raise 'failed to install deps' unless Apt.install(%w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations kde-release-keyring])
-
+    @deps = %w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations]
+      @deps << 'kde-release-keyring' unless ENV.fetch('JOB_NAME').include?('kde-release-keyring')
```
Comment (hsitter) [inline nci/lib/setup_repo.rb:63]: These two lines are incorrectly indented

File: nci/lib/setup_repo.rb [hunk 6]
```diff
@@ -59,8 +59,11 @@ def setup_repo!(with_source: false, with_proxy: true, with_install: true)
     # Make sure we have the latest pkg-kde-tools, not whatever is in the image.
     return unless with_install
 
-    raise 'failed to install deps' unless Apt.install(%w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations kde-release-keyring])
-
+    @deps = %w[pkg-kde-tools pkg-kde-tools-neon debhelper cmake quilt dh-python dh-translations]
```
Comment (hsitter) [inline nci/lib/setup_repo.rb:62]: this should be `deps` not `@deps`.

`@` denotes an instance variable. i.e. it is available in the entire Object, which isn't necessary here. what we need is a variable that is limited to the local scope

Comment (carlosdem) [reply nci/lib/setup_repo.rb]: thanks for your patience ...

**Pull request merged**